Hw05

Xokyopo opened the pull request.

Xokyopo committed:
[init] скопировал все файлы из "https://github.com/OtusTeam/Spring/tree/master/templates/hw05-jdbc-simple" в модуль "hw05"

Xokyopo committed:
[feat] добавил игнорирование любых логов в ".gitignore"

Xokyopo committed:
[feat] Добавил поле "jdbcTemplate" в класс "JdbcAuthorRepository" для доступа к данным

Xokyopo committed:
[feat] Реализовал "RowMapper" для клсса "Author" в репозитории "JdbcAuthorRepository.java"

Xokyopo committed:
[feat] Добавил константу "ROW_MAPPER" в класс "JdbcAuthorRepository" для хранения конвертера строк модели "Author"

Xokyopo committed:
[feat] написал реализацию метода "findAll" класса "JdbcAuthorRepository"

Xokyopo committed:
[feat] добавил реализацию метода "findById" класса "JdbcAuthorRepository"

Xokyopo committed:
[feat] добавил параметр "jdbcTemplate" в класс "JdbcGenreRepository"

Xokyopo committed:
[feat] добавил параметр "ROW_MAPPER" в класс "JdbcGenreRepository" для хранения конвертера строк модели Genre

Xokyopo committed:
[feat] Добавил реализацию метода "mapRow" во вложенный класс "GnreRowMapper" класса "JdbcGenreRepository"

Xokyopo committed:
[fix] Исправил ошибку наименования вложенного класса переименовав его с "GnreRowMapper" на "GenreRowMapper" в классе "JdbcGenreRepository"

Xokyopo committed:
[feat] добавил реализацию метода "findAll" класса "JdbcGenreRepository"

Xokyopo committed:
[feat] добавил реализацию метода "findById" класса "JdbcGenreRepository"

Xokyopo committed:
[feat] добавил параметр "jdbcTemplate" в класс "JdbcBookRepository"

Xokyopo committed:
[feat] Добавил константу "ROW_MAPPER" в класс "JdbcBookRepository" для хранения конвертера строк модели "Book"

Xokyopo committed:
[feat] Добавил реализацию метода "mapRow" во вложенном класс "BookRowMapper" класса "JdbcBookRepository"

Xokyopo committed:
[feat] Добавил реализацию метода "findAll" в классе "JdbcBookRepository"

Xokyopo committed:
[feat] добавил реализацию метода "findById" класса "JdbcBookRepository"

Xokyopo committed:
[feat] добавил реализацию метода "deleteById" класса "JdbcBookRepository"

Xokyopo committed:
[feat] добавил реализацию метода "update" класса "JdbcBookRepository"

Xokyopo committed:
[feat] добавил реализацию метода "insert" класса "JdbcBookRepository"

Xokyopo committed:
[fix] исправил ошибку, когда не найден объект по Ид в методе "deleteById" класса "JdbcBookRepository"

Xokyopo committed:
[fix] исправил ошибку, когда не найден объект по Ид в методе "deleteById" класса "JdbcAuthorRepository"

Xokyopo committed:
[fix] исправил ошибку, когда не найден объект по Ид в методе "deleteById" класса "JdbcGenreRepository"

Xokyopo committed:
[style] добавил пустую строку в конце файла JdbcBookRepositoryTest.java

vlad-iv commented on hw05/src/main/java/ru/otus/hw/repositories/JdbcAuthorRepository.java:
```diff
@@ -0,0 +1,45 @@
+
+        List<Author> authors = jdbcTemplate.query(queryString, queryParameters, ROW_MAPPER);
+
+        return !authors.isEmpty() ? Optional.of(authors.get(0)) : Optional.empty();
```
можно применить .stream().findFirst()

vlad-iv commented on hw05/src/main/java/ru/otus/hw/repositories/JdbcBookRepository.java:
```diff
@@ -0,0 +1,140 @@
+                    LEFT JOIN authors on authors.id = books.author_id
+                    LEFT JOIN genres on genres.id = books.genre_id
+                WHERE books.id = :id
+                """.replaceAll("\\s+", " ");
```
можно удалить replaceAll

vlad-iv commented on hw05/src/main/java/ru/otus/hw/repositories/JdbcBookRepository.java:
```diff
@@ -0,0 +1,140 @@
+
+        @Override
+        public Book mapRow(ResultSet rs, int rowNum) throws SQLException {
+            Author author = rs.getLong("author_id") > 0
```
можно удалить проверку на >0
и сделать жанр с автором обязательным

vlad-iv commented on hw05/src/main/resources/data.sql:
```diff
@@ -0,0 +1,14 @@
+insert into authors(full_name)
```
можно заменить тестовые данные на реальные или удалить)

Xokyopo committed:
[fix] Заменил получение по индексу на получение через стрим в методе "findById" в классе "JdbcAuthorRepository"

Но это однозначно медленней, чем простое получение по индексу еще и ошибку может высыпать

Xokyopo committed:
[fix] Заменил получение по индексу на получение через стрим в методе "findById" в классе "JdbcBookRepository"

Но это однозначно медленней, чем простое получение по индексу еще и ошибку может высыпать

Xokyopo committed:
[fix] Заменил получение по индексу на получение через стрим в методе "findById" в классе "JdbcGenreRepository"

Но это однозначно медленней, чем простое получение по индексу еще и ошибку может высыпать

Xokyopo committed:
[fix] удалил метод очистки лишник пустых символов в строках запросов в классе "JdbcBookRepository"

На самом деле они решали проблему связанную с появлением неизвестног осимола в строке, не знаю откуда он взялся но из за него не выполнялись запросы, было похоже что подмело первод строки. Сейчас тестил проблем небыло, откуда взялся тогда ошибка я хз.

Xokyopo committed:
[fix] заменил на реальные данные данные для разового импорта в "data.sql"

Чему это должно было меня научить, ведь это фал предоставил преподаватель, который не советовал менять лишнего?
